### PR TITLE
[ONNX] Support optional outputs for ONNX nodes

### PR DESCRIPTION
**Pull request by mbrookhart**
Also enables some cast tests that were incorrectly skipped.

### CODE REVIEW

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -3202,6 +3202,24 @@ def from_onnx(self, graph, opset, get_output_expr=False):
...
             else:
                 outputs_num = len(op)
+            if outputs_num > 1:
+                valid_outputs = [False] * outputs_num
+                for i in range(len(node_output)):
+                    if node_output[i] != "":
+                        valid_outputs[i] = True
+                if not all(valid_outputs):
+                    tup = op.astuple()
+                    if isinstance(tup, _expr.Tuple):
```
Comment (jwfromm) [inline python/tvm/relay/frontend/onnx.py:3212]: Does `op.astuple` not always return an `_expr.Tuple`?

**Review by jwfromm — APPROVED** [review]

**Comment by jwfromm** [comment]
LGTM, i think a few comments explaining the logic here would be helpful though.

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (mbrookhart) [reply python/tvm/relay/frontend/onnx.py:3212]: No, we use TupleWrapper for other nodes, like call nodes with multiple outputs, see split: https://github.com/apache/tvm/blob/461d06eb5cfc7954f1983779acd05c47cea269f1/python/tvm/relay/op/transform.py#L861

**Review by yzhliu — APPROVED** [review]
Thanks @mbrookhart @jwfromm

**Pull request merged**